### PR TITLE
updated to create regular churros and crunch.js sound location

### PR DESCRIPTION
**Pull request by elisa-lj11**
## Test Plan
1. Go to hifi://mexico/506.479,498.46,496.46/0,-0.743432,0,0.668812 (in front of churro stand)
2. Click on churro stand
3. A churro should appear
4. Drag the churro to your mouth
5. You should hear a crunch and the churro will disappear

Hopefully fixes https://highfidelity.fogbugz.com/f/cases/13902/Mystery-Eating-crunching-Churro-sounds-in-Mexico-near-fountain (won't know until more bug reports come up)

### CODE REVIEW

File: DomainContent/Mexico/createChurro.js [hunk 1]
```diff
@@ -47,7 +47,7 @@
     };
 
     this.mousePressOnEntity = function(entityID, mouseEvent) {
-        Entities.addEntity(churro, true);
+        Entities.addEntity(churro);
     };
 
 });
```
Comment (dback2) [inline DomainContent/Mexico/createChurro.js]: New line at end of file

File: DomainContent/Mexico/crunch.js [hunk 2]
```diff
@@ -10,39 +10,35 @@
...
-    _this.unload = function (entityID) {
+
+    _this.unload = function(entityID) {
+        if (injector) {
+            injector.stop();
+            injector = null;
+        }
         Script.update.disconnect(checkIfNearHead);
     };
 });
```
Comment (dback2) [inline DomainContent/Mexico/crunch.js]: New line at end of file

**Comment by YuppyPlays** [comment]
Testing.

**Comment by YuppyPlays** [comment]
Testing complete.

The crunch sound does play though it seems to be distorted / broken audio, sounds very glitchy, Tried this multiple times and the sound was the same every time.

**Fail** unless the sound is considered correct?.. Can record a soundbyte if needed.

**Comment by elisa-lj11** [comment]
@YuppyPlays could you record the soundbyte please?

**Comment by elisa-lj11** [comment]
I believe the issue is that the function that checks whether the churro is near an avatar's mouth (and triggers the eating sound) gets called multiple times before the entity can be deleted, so multiple audio injectors are playing over each other (the distorted sound).

File: DomainContent/Mexico/crunch.js [hunk 3]
```diff
@@ -8,41 +8,37 @@
...
+        injector = Audio.playSound(CRUNCH, {volume: VOLUME, position: MyAvatar.getJointPosition("Head")});
         Entities.deleteEntity(_this.entityID);
     };
 
-    // Helper function to see if the object is close to us
-    var isWithin10cm = function (val1, val2) {
-        if (Math.abs(Math.abs(val1) - Math.abs(val2)) <= WITHIN_10_CM) {
-            return true;
-        }
+    checkInterval = Script.setInterval(checkIfNearHead, INTERVAL);
```
Comment (dback2) [inline DomainContent/Mexico/crunch.js]: might be cleaner to put this in preload

**Comment by elisa-lj11** [comment]
@YuppyPlays ready for retesting

**Comment by YuppyPlays** [comment]
Testing.

**Comment by YuppyPlays** [comment]
**Testing complete:**

Churro sound now seems to play twice very fast like an echo, though it does sound much better than before.

**Fail** Unless this sound is considered okay, definitely an improvement over the previous.

Soundbyte of this in .zip:
[Churro sound.zip](https://github.com/highfidelity/hifi-content/files/2007687/Churro.sound.zip)

**Comment by elisa-lj11** [comment]
@YuppyPlays I have updated the script, I don't believe there is a possibility of overlapping crunch sounds anymore. Here is the original sound file for reference.
[Apple_Bite-Simon_Craggs-1683647397.zip](https://github.com/highfidelity/hifi-content/files/2014500/Apple_Bite-Simon_Craggs-1683647397.zip)

**Comment by YuppyPlays** [comment]
Testing.

**Comment by YuppyPlays** [comment]
**Testing Complete:**

Churros now work perfectly and sound just like the sound sent by Elisa every time.

**Ate over 25 Churros to test this, thank god avatars don't gain weight**

**Pull request merged**